app/src/main/java/com/kickstarter/viewmodels/projectpage/AddOnsViewModel.kt

```diff
@@ -169,25 +170,32 @@ class AddOnsViewModel(val environment: Environment, bundle: Bundle? = null) : Vi
     }
 
     private fun getAddOns(selectedShippingRule: ShippingRule) {
-        scope.launch(dispatcher) {
-            apolloClient
-                .getProjectAddOns(
-                    slug = project.slug() ?: "",
-                    locationId = selectedShippingRule.location() ?: LocationFactory.empty()
-                )
-                .asFlow()
-                .onStart {
-                    emitCurrentState(isLoading = true)
-                }
-                .map { addOns ->
-                    if (!addOns.isNullOrEmpty()) {
-                        this@AddOnsViewModel.addOns = getUpdatedList(addOns, backedAddOns)
+        // - Do not execute call unless reward has addOns
+        if (currentUserReward.hasAddons() || backing?.addOns().isNotNull()) {
```

Comment:
Not related with filtering, small bug on rewards without addOns making a network request